### DISCUSSION TITLE
screensharing is very resource intensive

### DISCUSSION THREAD

**Discussion by MangoIV** [YaLTeR/niri#675] [Q&A]
I was wondering how niri is involved with screensharing and what could be the bottle necks there ~ for some reason screen sharing for me is really resource intensive to the point where every task feels really slow ~ the screen I'm usually sharing *is* a 4k screen and the machine is *just* a laptop (but a rather recent one) so it's possible that the laptop is just underspecced for this kinda task ~ but I really don't even know how to e.g. trouble shoot what *could* be slow / if I use graphics acceleration for screen sharing etc. 

thanks in advance <3

**Comment by YaLTeR** [comment]
What program are you using for screensharing? Is it using the screencast portal or wlr-screencopy?

**Reply by MangoIV** [reply]
it is using the gnome desktop portal, so the former, I think

**Reply by YaLTeR** [reply]
By resource intensive do you mean your niri framerate drops, or something else?

Also to diagnose this, I'll need a Tracy recording. This involves building niri with `cargo build --release --features=profile-with-tracy-ondemand` and building tracy-capture from https://github.com/wolfpld/tracy/tree/v0.11.1, then recording for some seconds while you're screencasting.

**Reply by MangoIV** [reply]
> By resource intensive do you mean your niri framerate drops, or something else?

yea not only niri but also browsers (or any graphical application for the matter) get very slow 

> Also to diagnose this, I'll need a Tracy recording. This involves building niri with cargo build --release --features=profile-with-tracy-ondemand and building tracy-capture from https://github.com/wolfpld/tracy/tree/v0.11.1, then recording for some seconds while you're screencasting.

thank you I will do this as soon as I find time 

Thank you so much for your kind help until now <3

**Reply by YaLTeR** [reply]
Also it'd be helpful if you could try screencasting in GNOME Shell and see if that works better. It has a GPU optimization that niri doesn't so that could help pinpoint the issue a bit.

**Reply by MangoIV** [reply]
good idea ~ I'll see :)
